java-samples/corda5-negotiation-cordapp/workflows/src/test/java/com/r3/developers/samples/negotiation/workflows/NegotiationFlowTests.java

```diff
@@ -0,0 +1,302 @@
+package com.r3.developers.samples.negotiation.workflows;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.r3.developers.samples.negotiation.workflows.accept.AcceptFlowArgs;
+import com.r3.developers.samples.negotiation.workflows.accept.AcceptFlowRequest;
+import com.r3.developers.samples.negotiation.workflows.modify.ModifyFlowArgs;
+import com.r3.developers.samples.negotiation.workflows.modify.ModifyFlowRequest;
+import com.r3.developers.samples.negotiation.workflows.propose.ProposalFlowArgs;
+import com.r3.developers.samples.negotiation.workflows.propose.ProposalFlowRequest;
+import net.corda.testing.driver.AllTestsDriver;
+import net.corda.testing.driver.DriverNodes;
+import net.corda.v5.base.types.MemberX500Name;
+import net.corda.v5.ledger.utxo.UtxoLedgerService;
+import net.corda.v5.application.flows.CordaInject;
+import net.corda.virtualnode.VirtualNodeInfo;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class NegotiationFlowTests {
+    private static final Logger logger = LoggerFactory.getLogger(NegotiationFlowTests.class);
+    private static final MemberX500Name alice = MemberX500Name.parse("CN=Alice, OU=Application, O=R3, L=London, C=GB");
+    private static final MemberX500Name bob = MemberX500Name.parse("CN=Bob, OU=Application, O=R3, L=London, C=GB");
+    private static final MemberX500Name charles = MemberX500Name.parse("CN=Charles, OU=Application, O=R3, L=London, C=GB");
+    private Map<MemberX500Name, VirtualNodeInfo> virtualNodes;
+    @CordaInject
+    UtxoLedgerService utxoLedgerService;
+    private static final ObjectMapper jsonMapper;
+
+    static {
+        jsonMapper = new ObjectMapper();
+        SimpleModule module = new SimpleModule();
+        module.addSerializer(MemberX500Name.class, new MemberX500NameSerializer());
+        module.addDeserializer(MemberX500Name.class, new MemberX500NameDeserializer());
+        jsonMapper.registerModule(module);
+    }
+
+    // Create the driver with Alice and Bob as participants, and Charlie as the notary
+    // This is to be created for all tests
+    @RegisterExtension
+    private final AllTestsDriver driver =
+            new DriverNodes(alice, bob, charles).withNotary(charles, 1).forAllTests();
```

Comment:
Should charles appear twice? I think if you pass him in the withNotary you don't need to have him in DriverNodes

Reply:
Removed Charlie from nodes :)